File: tests/comparison/cpp/loop_intensive.cpp
#include <iostream>
#include <cstdlib>

/**
 * Loop-intensive benchmark for C++
 * Usage: ./loop_intensive <iterations>
 */

long long nested_loops(int n) {
    // Nested loop computation - standardized across all languages
    long long total = 0;
    int i = 0;
    while (i < n) {
        int j = 0;
        while (j < 100) {
            total += i * j;
            j++;
        }
        i++;
    }
    return total;
}

long long simple_loop(int n) {
    // Simple loop with arithmetic - standardized across all languages
    long long total = 0;
    int i = 0;
    while (i < n) {
        total += i * i;
        i++;
    }
    return total;
}

int main(int argc, char* argv[]) {
    if (argc != 2) {
        std::cerr << "Usage: " << argv[0] << " <iterations>" << std::endl;
        return 1;
    }
    
    int n = std::atoi(argv[1]);
    if (n < 0) {
        std::cerr << "Error: iterations must be non-negative" << std::endl;
        return 1;
    }
    
    // Choose algorithm based on scale
    long long result;
    if (n <= 100000) {
        result = nested_loops(n / 100);  // Scale down for nested loops
    } else {
        result = simple_loop(n);
    }
    
    // Don't print result to avoid affecting timing
    
    return 0;
}

File: tests/comparison/python/loop_intensive.py
#!/usr/bin/env python3
"""
Loop-intensive benchmark for Python
Usage: python loop_intensive.py <iterations>
"""

import sys

def nested_loops(n):
    """Nested loop computation - standardized across all languages"""
    total = 0
    i = 0
    while i < n:
        j = 0
        while j < 100:
            total += i * j
            j += 1
        i += 1
    return total

def simple_loop(n):
    """Simple loop with arithmetic - standardized across all languages"""
    total = 0
    i = 0
    while i < n:
        total += i * i
        i += 1
    return total

def main():
    if len(sys.argv) != 2:
        print("Usage: python loop_intensive.py <iterations>")
        sys.exit(1)
    
    try:
        n = int(sys.argv[1])
    except ValueError:
        print("Error: iterations must be an integer")
        sys.exit(1)
    
    if n < 0:
        print("Error: iterations must be non-negative")
        sys.exit(1)
    
    # Choose algorithm based on scale
    if n <= 100000:
        result = nested_loops(n // 100)  # Scale down for nested loops
    else:
        result = simple_loop(n)
    
    # Don't print result to avoid affecting timing

if __name__ == "__main__":
    main()